mautic/mautic
# Add Bitly bundle

closes https://github.com/mautic/mautic/issues/8471


| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | I'd say yes as far as we fix an old broken service
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes #8471



#### Description:

We build a new shortener service interface for Mautic https://github.com/mautic/mautic/pull/12299.  
Would be good to add the most used shortener service to core and others could users install manually/or by marketplace. 
This PR add bitly support to core



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  You need use https://github.com/mautic/mautic/pull/12299 first
3. YInstall bitly (hope one from core later)
4. Get access key from [app.bitly.com/settings/api](https://app.bitly.com/settings/api/) and setup/enable it
5. Go to configuration and choose bitly as shortener service
6. Go to SMS and try send SMS with URL. You should see bitly shortener URL
